src/modern/getQueryName.js

```diff
@@ -6,7 +6,9 @@ export default function getQueryName(route) {
     }
 
     if (typeof query === 'function') {
-      return query().name;
+      query = query();
+      if (query.name) return query.name;
+      if (query.params && query.params.name) return query.params.name;
```

Comment:
@xdissent out of curiosity, what triggers this?

Reply:
ah, i see... some cruft on my part. fixing it in https://github.com/relay-tools/found-relay/pull/170